lib/sqlalchemy/engine/result.py

```diff
@@ -177,6 +182,29 @@ def _row_as_tuple_getter(
         indexes = self._indexes_for_keys(keys)
         return tuplegetter(*indexes)
 
+    def _make_key_to_index(
+        self, keymap: Mapping[_KeyType, Sequence[Any]], index: int
+    ) -> Mapping[_KeyType, int]:
+        return {
+            key: rec[index]
```

Comment:
**mike bayer** (zzzeek) wrote:

great, this is great

seemed like there was something more complicated here previously...

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4575